fiona/fio/helpers.pyi

```diff
@@ -0,0 +1,27 @@
+from click.testing import _NamedTextIOWrapper
+from typing import (
+    Any,
+    Dict,
+    Iterator,
+    List,
+    Optional,
+    Tuple,
+    Union,
+)
+from fiona.model import Geometry
+
+
+def eval_feature_expression(
+    feature: Dict[str, Union[Dict[str, Union[str, List[List[List[float]]]]], str, Dict[str, Optional[Union[float, str, int]]]]],
```

Comment:
For complex combinations of basic types, it's probably useful to introduce names for (sub)expressions.

I like the style that the httpx library uses:
https://github.com/encode/httpx/blob/master/httpx/_client.py#L1111